public/add-to-storage.js

```diff
@@ -0,0 +1,63 @@
+"use strict";
+
+var addAnswersToSessionStorage = function(){
+  var nameInput = document.getElementsByClassName("name-input")[0];
+  var ageInput = document.getElementsByClassName("age-input")[0];
+  var strange = document.getElementById("strange");
+  var happy = document.getElementById("happy");
+  var angry = document.getElementById("angry");
+  var fun = document.getElementById("fun");
+  var boring = document.getElementById("boring");
+  var kind = document.getElementById("kind");
+  var sad = document.getElementById("sad");
+  var happyEmoji = document.getElementsByClassName("emoji")[0];
+  var sadEmoji = document.getElementsByClassName("emoji")[1];
+  var indifferentEmoji = document.getElementsByClassName("emoji")[2];
+  var personality = [];
+
+  addKeyupEvent('name', nameInput);
+  addKeyupEvent('age', ageInput);
+
+  addClickEventArray('personality', strange, 'strange', personality);
+  addClickEventArray('personality', happy, 'happy', personality);
+  addClickEventArray('personality', angry, 'angry', personality);
+  addClickEventArray('personality', fun, 'fun', personality);
+  addClickEventArray('personality', boring, 'boring', personality);
+  addClickEventArray('personality', kind, 'kind', personality);
+  addClickEventArray('personality', sad, 'sad', personality);
+
+  addClickEventSingle('feelings', sadEmoji, 'sad');
+  addClickEventSingle('feelings', happyEmoji, 'happy');
+  addClickEventSingle('feelings', indifferentEmoji, 'indifferent');
+
+  function addKeyupEvent(key, element){
+    if(!element){ return; };
+    element.addEventListener("keyup", function(){
+      addToStorageValue(key, element.value)
+    });
+  };
+
+  function addClickEventSingle(key, element, value){
+    if(!element){ return; };
+    element.addEventListener("click", function(){
+      addToStorageValue(key, value)
+    });
+  }
+
+  function addClickEventArray(key, element, value, array){
+    if(!element){ return; };
+    element.addEventListener("click", function(){
+      addToStorageArray(key, value, array);
+    });
+  };
+
+  function addToStorageValue(key, value){
+    sessionStorage.setItem(key, value);
+  };
+
+  // JSON.stringify() is used here as session storage only supports strings
+  function addToStorageArray(key, value, array){
+    array.push(value);
```

Comment:
I think it would be better to separate the 2 steps
 1. mutate the array, appending elem
 2. serialise and store